src/main/java/io/odpf/depot/bigquery/handler/JsonErrorHandler.java

```diff
@@ -38,55 +59,49 @@ public void handle(Map<Long, List<BigQueryError>> insertErrors, List<Record> rec
         FieldList existingFieldList = schema.getFields();
         List<Entry<Long, List<BigQueryError>>> unknownFieldBqErrors = getUnknownFieldBqErrors(insertErrors);
         if (!unknownFieldBqErrors.isEmpty()) {
-            Set<Field> missingFields = unknownFieldBqErrors
+            ArrayList<Field> bqSchemaFields = unknownFieldBqErrors
                     .parallelStream()
                     .map(x -> getColumnNamesForRecordsWhichHadUknownBqFieldErrors(records, x))
-                    .flatMap(keys -> keys.stream())
+                    .flatMap(Collection::stream)
                     .filter(key -> filterExistingFields(existingFieldList, key))
-                    .map(key -> getField(key))
-                    .collect(Collectors.toSet());
-            ArrayList<Field> bqSchemaFields = new ArrayList<>(missingFields);
+                    .map(this::getField).distinct().collect(Collectors.toCollection(ArrayList::new));
```

Comment:
@lavkesh we need to revert this fix, the reason we had set is to remove duplicates, I see we have added "distinct" to achieve the same , but the documentation of distinct says "For ordered streams, the selection of distinct elements is stable (for duplicated elements, the element appearing first in the encounter order is preserved.) For unordered streams, no stability guarantees are made." . Given  the stream here is unordered  this will lead to duplicates. So we need to revert to set.

Reply:
ah. i am not sure what you mean. 
if it's unordered, the document is saying that it may not guarantee which object distinct will choose..
how does it matter if the equals are same.. 
in this case unknownFieldBqErrors and its operations are ordered.. It's a list.